contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -1939,6 +1959,35 @@ check_nullable_identity_constraint(RangeVar *relation, ColumnDef *column)
 						relation->relname)));
 }
 
+/*
+ * Returns true iff the function is a system function. For Babelfish, this means searching in the "sys"
+ * schema, but the "sys" schema is meant to be opaque to customers, so to match SQL Server behavior
+ * we assume that it will never be schema-qualified in order to be true.
+ */
+static bool
+isTsqlSystemFunc(FuncCall *fc)
+{
+	List *name_to_search;
+	ObjectWithArgs *owa = makeNode(ObjectWithArgs);
+	if (list_length(fc->funcname) == 1)
+	{
+		/* explicitly search in the "sys" schema */
+		name_to_search = list_make2(makeString("sys"), linitial(fc->funcname));
+	}
+	else
+	{
+		return false;
+	}
+
+	owa->objname = name_to_search;
+	owa->args_unspecified = true;
+
+	if (LookupFuncWithArgs(OBJECT_FUNCTION, owa, true))
```

Comment:
how expensive is this call? I *think* it should be fine assuming it's just syscache lookups, since this would only be called in very rare cases when we need to actually check a column default function on a temp table.

Reply:
I don't know how expensive this call is, but as you said, this is buried below sufficient levels of checking that it shouldn't come up very often in the code path anyways. Basically, this would only get evaluated if we are dealing with a temp table using a function for a column default, and the function call is not schema-qualified. I suppose it could be made more efficient if we generated a list of all possible system functions and did a direct comparison against that list, but that seems like a lot of work.